pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -491,3 +539,178 @@ func (hc *HealthChecker) checkRemoteClusterGatewaysHealth(ctx context.Context) e
+			if err != nil {
+				nonExistentGatewayMirrors = append(nonExistentGatewayMirrors, fmt.Sprintf("%s.%s", gatewayMirrorName, svc.Annotations[k8s.RemoteGatewayNsLabel]))
+			}
+		}
+
+		if len(nonExistentGatewayMirrors) > 0 {
+			return fmt.Errorf("Some mirrored gateway services of mirrored services don't exist:\n\t%s", strings.Join(nonExistentGatewayMirrors, "\n\t"))
+		}
+		return nil
+	}
+	return &SkipError{Reason: "not checking muticluster"}
+}
+
+func (hc *HealthChecker) checkIfGatewaysHaveExternalIP() error {
+	if hc.Options.ShouldCheckMulticluster {
+
+		var gatewaysWithNoExternalIps []string
+		services, err := hc.kubeAPI.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, svc := range services.Items {
+			if gatewayService(svc) {
+				// check if there is an external IP for the gateway service
+				if len(svc.Status.LoadBalancer.Ingress) <= 0 {
+					gatewaysWithNoExternalIps = append(gatewaysWithNoExternalIps, fmt.Sprintf("%s.%s", svc.Name, svc.Namespace))
+				}
+			}
+		}
+
+		if len(gatewaysWithNoExternalIps) > 0 {
+			return fmt.Errorf("Some gateways don't have external Ips:\n\t%s", strings.Join(gatewaysWithNoExternalIps, "\n\t"))
+		}
+		return nil
+	}
+	return &SkipError{Reason: "not checking muticluster"}
+}
+
+func gatewayService(service corev1.Service) bool {
+	_, hasGtwName := service.Annotations[k8s.MulticlusterGatewayAnnotation]
+	return hasGtwName
+}
+
+func (hc *HealthChecker) checkIfAllGatewaysHaveCorrectPortNames() error {
+	if hc.Options.ShouldCheckMulticluster {
+
+		var gatewaysWithMisconfiguredPorts []string
+		services, err := hc.kubeAPI.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, svc := range services.Items {
+			if gatewayService(svc) {
+				// check if the gateway service has relevant ports
+				portNames := []string{k8s.GatewayPortName, k8s.ProbePortName}
+				for _, portName := range portNames {
+					if !ifPortExists(svc.Spec.Ports, portName) {
+						gatewaysWithMisconfiguredPorts = append(gatewaysWithMisconfiguredPorts, fmt.Sprintf("port %s does not exist for gateway %s.%s", portName, svc.Name, svc.Namespace))
+					}
+				}
+			}
+		}
+
+		if len(gatewaysWithMisconfiguredPorts) > 0 {
+			return fmt.Errorf("Some gateway's have misconfigured ports:\n\t%s", strings.Join(gatewaysWithMisconfiguredPorts, "\n\t"))
+		}
+		return nil
+	}
+	return &SkipError{Reason: "not checking muticluster"}
+}
+
+func ifPortExists(ports []corev1.ServicePort, portName string) bool {
+	for _, port := range ports {
+		if port.Name == portName {
+			return true
+		}
+	}
+
+	return false
+}
+func (hc *HealthChecker) checkifGatewaysExistforExportedServices() error {
+	if hc.Options.ShouldCheckMulticluster {
+
+		var nonExistentGateways []string
+		exportedServices, err := hc.kubeAPI.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{})
```

Comment:
this variable name is misleading since this is all services, not just the exported ones.